test/math.jl

```diff
@@ -842,24 +842,140 @@ end
         @test atan2(-T(1235.2341234), -T(2.5)) === -(T(pi)-(atan(abs(-T(1235.2341234)/T(2.5)))-_ATAN2_PI_LO(T)))
     end
 end
+#prev, current, next float
+pcnfloat(x) = prevfloat(x), x, nextfloat(x)
+@testset "sinh" begin
+    for T in (Float32, Float64)
+        @test sinh(zero(T)) === zero(T)
+        @test sinh(-zero(T)) === -zero(T)
+        @test sinh(nextfloat(zero(T))) === nextfloat(zero(T))
+        @test sinh(prevfloat(zero(T))) === prevfloat(zero(T))
+        @test sinh(T(1000)) === T(Inf)
+        @test sinh(-T(1000)) === -T(Inf)
+        @test isnan_type(T, sinh(T(NaN)))
+    end
+    for x in (pcnfloat(2.0^-28)..., pcnfloat(22.0)..., pcnfloat(709.7822265633563)...)
+        by = sinh(big(x))
+        @test Float64((sinh(x) - by))/eps(abs(Float64(by))) <= 1
+        bym = sinh(big(-x))
+        @test Float64(abs(sinh(-x) - bym))/eps(abs(Float64(bym))) <= 1
+    end
+    for x in (pcnfloat(2f-12)..., pcnfloat(9f0)..., pcnfloat(88.72283f0)...)
+        by = sinh(big(x))
+        @test Float32((sinh(x) - by))/eps(abs(Float32(by))) <= 1
+        bym = sinh(big(-x))
+        @test Float32(abs(sinh(-x) - bym))/eps(abs(Float32(bym))) <= 1
+    end
+end
 
-@testset "acos #23283" begin
+@testset "cosh" begin
     for T in (Float32, Float64)
-        @test acos(zero(T)) === T(pi)/2
-        @test acos(-zero(T)) === T(pi)/2
-        @test acos(nextfloat(zero(T))) === T(pi)/2
-        @test acos(prevfloat(zero(T))) === T(pi)/2
-        @test acos(one(T)) === T(0.0)
-        @test acos(-one(T)) === T(pi)
-        for x in (0.45, 0.6, 0.98)
-            by = acos(big(T(x)))
-            @test T((acos(T(x)) - by))/eps(abs(T(by))) <= 1
-            bym = acos(big(T(-x)))
-            @test T(abs(acos(T(-x)) - bym))/eps(abs(T(bym))) <= 1
-        end
-        @test_throws DomainError acos(-T(Inf))
-        @test_throws DomainError acos(T(Inf))
-        @test isnan_type(T, acos(T(NaN)))
+        @test cosh(zero(T)) === one(T)
+        @test cosh(-zero(T)) === one(T)
+        @test cosh(nextfloat(zero(T))) === one(T)
+        @test cosh(prevfloat(zero(T))) === one(T)
+        @test cosh(T(1000)) === T(Inf)
+        @test cosh(-T(1000)) === T(Inf)
+        @test isnan_type(T, cosh(T(NaN)))
+    end
+    for x in (pcnfloat(2.7755602085408512e-17)..., pcnfloat(22.0)..., pcnfloat(709.7822265633563)...)
+        by = cosh(big(x))
+        @test Float64((cosh(x) - by))/eps(abs(Float64(by))) <= 1
```

Comment:
Would be clearer to use `@test cosh(x) ≈ by rtol=eps(Float64)` here and similarly in the other tests.

Reply:
`big(y)` is meant to be `big(x)`, right?